[2024-12-25T15:32:57.307+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-25T15:32:57.546+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T14:00:00+00:00 [queued]>
[2024-12-25T15:32:57.599+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T14:00:00+00:00 [queued]>
[2024-12-25T15:32:57.600+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2024-12-25T15:32:57.705+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): fetch_and_produce_data> on 2024-12-25 14:00:00+00:00
[2024-12-25T15:32:57.733+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'api_to_kafka_pipeline', 'fetch_and_produce_data', 'scheduled__2024-12-25T14:00:00+00:00', '--job-id', '54', '--raw', '--subdir', 'DAGS_FOLDER/api_to_kafka_dag.py', '--cfg-path', '/tmp/tmp6bgbh3sq']
[2024-12-25T15:32:57.755+0000] {standard_task_runner.py:105} INFO - Job 54: Subtask fetch_and_produce_data
[2024-12-25T15:32:57.750+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=379) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-25T15:32:57.757+0000] {standard_task_runner.py:72} INFO - Started process 383 to run task
[2024-12-25T15:32:57.950+0000] {task_command.py:467} INFO - Running <TaskInstance: api_to_kafka_pipeline.fetch_and_produce_data scheduled__2024-12-25T14:00:00+00:00 [running]> on host 0f607e57b1dd
[2024-12-25T15:32:58.548+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='api_to_kafka_pipeline' AIRFLOW_CTX_TASK_ID='fetch_and_produce_data' AIRFLOW_CTX_EXECUTION_DATE='2024-12-25T14:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-25T14:00:00+00:00'
[2024-12-25T15:32:58.556+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-25T15:32:59.516+0000] {logging_mixin.py:190} INFO - âœ… Data fetched successfully for IBM at 5min interval.
[2024-12-25T15:32:59.517+0000] {logging_mixin.py:190} INFO - ðŸš€ Sending data to Kafka...
[2024-12-25T18:02:22.031+0000] {job.py:229} INFO - Heartbeat recovered after 871.00 seconds
[2024-12-25T18:03:02.252+0000] {job.py:229} INFO - Heartbeat recovered after 34.75 seconds
[2024-12-26T00:12:46.974+0000] {job.py:229} INFO - Heartbeat recovered after 21082.00 seconds
[2024-12-26T05:36:43.950+0000] {job.py:229} INFO - Heartbeat recovered after 19431.73 seconds
[2024-12-26T05:37:05.522+0000] {local_task_job_runner.py:346} WARNING - State of this instance has been externally set to up_for_retry. Terminating instance.
[2024-12-26T05:37:05.588+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-12-26T05:37:05.627+0000] {process_utils.py:132} INFO - Sending 15 to group 383. PIDs of all processes in the group: [383]
[2024-12-26T05:37:05.644+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 383
[2024-12-26T05:38:05.675+0000] {process_utils.py:150} WARNING - process psutil.Process(pid=383, name='airflow task ru', status='sleeping', started='15:32:57') did not respond to SIGTERM. Trying SIGKILL
[2024-12-26T05:38:05.701+0000] {process_utils.py:87} INFO - Sending the signal 9 to group 383
[2024-12-26T05:38:05.760+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=383, name='airflow task ru', status='terminated', exitcode=<Negsignal.SIGKILL: -9>, started='15:32:57') (383) terminated with exit code -9
[2024-12-26T05:38:05.764+0000] {standard_task_runner.py:190} ERROR - ('Job 54 was killed before it finished (likely due to running out of memory)', 'For more information, see https://airflow.apache.org/docs/apache-airflow/stable/troubleshooting.html#LocalTaskJob-killed')
